packages/babel-traverse/src/path/introspection.ts

```diff
@@ -396,7 +399,10 @@ export function _guessExecutionStatusRelativeToDifferentFunctions(
   // then we can be a bit smarter and handle cases where the function is either
   // a. not called at all (part of an export)
   // b. called directly
-  const binding = target.scope.getBinding(target.node.id.name);
+  const bindingName = targetIsGenericFunctionExpression
+    ? target.parent["id"]["name"]
+    : target.node["id"]["name"];
```

Comment:
Still not 100% type checked, but this should be a bit better.

```suggestion
  const bindingName = targetIsGenericFunctionExpression
    ? (target.parent as t.VariableDeclarator).id.name
    : (target.node as t.FunctionDeclaration).id.name;
```